lib/vrl/stdlib/src/chunks.rs

```diff
@@ -0,0 +1,153 @@
+use ::value::Value;
+use vrl::prelude::*;
+
+const MAX_UTF8_CODE_POINT_LENGTH_IN_BYTES: usize = 4;
+
+fn chunks(value: Value, chunk_size: Value) -> Resolved {
+    let chunk_size = chunk_size.try_integer()? as usize;
+    let bytes = value.try_bytes()?;
+
+    let data_length = bytes.len();
+    let mut chunked_data: Vec<&[u8]> = Vec::new();
+
+    let mut backtrack = 0;
+    let mut start = 0;
+    let mut end;
+
+    if chunk_size < MAX_UTF8_CODE_POINT_LENGTH_IN_BYTES {
+        return Err(r#""chunk_size" must be greater than or equal to 4 bytes"#.into());
+    }
+
+    loop {
+        let is_last_chunk = (start < data_length) && (start + chunk_size >= data_length);
+
+        if is_last_chunk {
+            chunked_data.push(&bytes[start..]);
+            break;
+        }
+
+        end = start + chunk_size - backtrack;
+
+        if is_root_utf8_byte(&bytes[end]) {
+            chunked_data.push(&bytes[start..end]);
+            backtrack = 0;
+            start = end;
+        } else {
+            backtrack += 1;
+
+            if backtrack >= MAX_UTF8_CODE_POINT_LENGTH_IN_BYTES {
+                return Err("Bytes are not valid UTF-8".into());
+            }
+        }
+    }
```

Comment:
This is a very imperative way to chunk the data, but it made the most sense at the time. I would appreciate feedback on other - especially more functional - ways to approach it.